### PR TITLE
Proposal: stPagination - stCurrentPage attribute

### PR DESCRIPTION
**Pull request by keropodium**
I represent a company responsible for the tax administration of **several** spanish municipalities. Right now we are rewriting our old tax management application, *done in C++*, to a new and shining **Angular SPA**.
In the early phases of our development we tried several table components, like *ui-grid*, but yours ended being the one we truly needed. We enjoy it very much.

Our app lets the user have multiple *tasks* opened at the same time, each task having **several** reusable screens, and lets them move easily between them. Our screens store their *state* to communicate with other screens or simply to recover its previous state when the user returns.
Because of that, we need a very simple mechanism for always being aware of which one is the current page and also to change it *easily* programmatically via our controllers.

Basically I wrote a two-way data binding for the current page, a few tests and nothing more. This has been on testing for a week with no issues.

I know we could have ended writing a plugin for this, but after a long reunion, my superiors allowed me to code our solution and contribute to your project. Perhaps this is the first of many.

### CODE REVIEW

**Comment by Tronix117** [comment]
:+1: Should be merged

**Comment by jmorrispa** [comment]
:+1:  just about to post a pull request with the same feature

**Comment by rafaelhz** [comment]
There's a question about that on stack overflow with many views, and the solution is terrible, it should be merged.
http://stackoverflow.com/questions/29921339/smart-table-setting-page-from-code/32136115#32136115

**Comment by lorenzofox3** [comment]
The idea is that I want the current page to be readOnly which you can get by registering a callback whenever the page change (stPageChange). 

If you want to navigate to the page, you should explicitly call the ctrl.slice function which is what is done in the PR. 

However my experience has proven, that if you set a two way data bindings for that purpose, model tends to be unstable, people open tickets as things go out of control, and stPipe may be called multiple times.

I suggest that if you want this feature, you create you own directive which would be basically what the code above is 
1. watching a bound property
2. call ctrl.slice accordingly

But I don't want it to be part of the core

Thanks anyway for the PR

**Comment by Sadi-1992** [comment]
> [...]model tends to be unstable, people open tickets as things go out of control[...]

is see people opening tickets because of that missing feature for years and there is still no solution...
could you please provide a fiddle for a directive with an two-way-data-binding like 
`current-page="(SCOPE HERE TO ACCESS IN OWN CONTROLLER)"` ?

I am trying to solve this for 4 days now and its frustrating

**Comment by lorenzofox3** [comment]
you have an example on the documentation website ... linking to [live running sample](https://plnkr.co/edit/LB2f9P?p=info)
